build_a_container.rst

```diff
@@ -270,6 +270,244 @@ containers. The containers are decrypted at runtime entirely in kernel space,
+   INFO:    Using crun runtime for buildkitd daemon.
+   INFO:    running buildkitd server on /run/user/1000/buildkit/buildkitd-8961170237105250.sock
+   [+] Building 15.1s (9/9)
+   [+] Building 15.2s (9/9) FINISHED
+   => [internal] load build definition from Dockerfile               0.0s
+   => => transferring dockerfile: 549B                               0.0s
+   => [internal] load metadata for docker.io/library/debian:latest   0.5s
+   => [internal] load .dockerignore                                  0.0s
+   => => transferring context: 2B                                    0.0s
+   => [1/5] FROM docker.io/library/debian:latest@sha256:fab22df3737  1.0s
+   => => resolve docker.io/library/debian:latest@sha256:fab22df3737  0.0s
+   => => extracting sha256:8457fd5474e70835e4482983a5662355d892d5f6  1.0s
+   => [2/5] RUN apt-get update                                       2.2s
+   => [3/5] RUN apt-get install -y cowsay lolcat                     7.9s
+   => [4/5] RUN echo $'#! /bin/bash \necho from cmdline: $@ | /usr/  0.1s
+   => [5/5] RUN chmod +x /myscript.sh                                0.1s
+   => exporting to docker image format                               3.3s
+   => => exporting layers                                            2.7s
+   => => exporting manifest sha256:fc7222347c207c35165ccd2fee562af9  0.0s
+   => => exporting config sha256:74c5da659e8504e4be283ad6d82774194e  0.0s
+   => => sending tarball                                             0.5s
+   Getting image source signatures
+   Copying blob 8457fd5474e7 done   |
+   Copying blob 4769fe2f22da done   |
+   Copying blob 173d009c20af done   |
+   Copying blob 7ec86debbe9b done   |
+   Copying blob 491c7ee403c2 done   |
+   Copying config 74b69e878e done   |
+   Writing manifest to image destination
+   INFO:    Converting OCI image to OCI-SIF format
+   INFO:    Squashing image to single layer
+   INFO:    Writing OCI-SIF image
+   INFO:    Cleaning up.
+   INFO:    Build complete: ./lolcow.oci.sif
+
+   $ echo "world" | singularity run --oci ./lolcow.oci.sif hello
+   _____________________
+   < from cmdline: hello >
+   ---------------------
+         \   ^__^
+            \  (oo)\_______
+               (__)\       )\/\
+                  ||----w |
+                  ||     ||
+   ___________________
+   < from stdin: world >
+   -------------------
+         \   ^__^
+            \  (oo)\_______
+               (__)\       )\/\
+                  ||----w |
+                  ||     ||
+
+{Singularity} uses `buildkit <https://docs.docker.com/build/buildkit/>`__ to
+build an OCI image from a Dockerfile. It checks if there is a ``buildkitd``
+daemon that is already running on the system (and whose permissions allow access
+by the current user), and if so, that daemon is used for the build process. If a
+usable ``buildkitd`` daemon is not found, {Singularity} will launch an
+emphemeral ``buildkitd`` daemon of its own, inside a :ref:`user namespace
```

Comment:
```suggestion
ephemeral ``buildkitd`` daemon of its own, inside a :ref:`user namespace
```

Reply:
Other thought here is that probably we see a `singularity` process and not a `buildkitd` process when our custom  buildkitd is running?

If that's the case it might be worth a note that the buildkitd is embedded with singularity. If we say a `buildkitd` is running then people will expect to see that in ps.